app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -291,105 +318,102 @@ export default {
+    },
+    init() {
+      // Since change is triggered during loading
+      this.setDirtyFlag(false);
+    },
+    // ----------------------------------------------------------------------------------/ FormIO Handlers
 
-          // Navigate back to navigation page on success
-          this.$router.push({
-            name: 'FormManage',
-            query: {
-              f: this.formId,
-            },
-          });
+    // ---------------------------------------------------------------------------------------------------
+    // Saving the Schema
+    // ---------------------------------------------------------------------------------------------------
+    async submitFormSchema() {
+      try {
+        this.saving = true;
+        // Once the form is done disable the "leave site/page" messages so they can quit without getting whined at
+        await this.setDirtyFlag(false);
 
-          // Draft version is now the latest - update route to reflect that
-          // this.$router.push({
-          //   name: 'FormDesigner',
-          //   query: {
-          //     f: this.formId,
-          //     v: this.draftId,
-          //   },
-          // });
-        } catch (error) {
-          this.addNotification({
-            message:
-              'An error occurred while attempting to update this form. If you need to refresh or leave to try again later, you can Export the existing design on the page to save for later.',
-            consoleError: `Error updating form ${this.formId} schema version ${this.versionId}: ${error}`,
-          });
-        }
-      } else {
-        // If creating a new form, add the form and then a version
-        try {
-          const emailList =
-            this.sendSubRecieviedEmail &&
-            this.submissionReceivedEmails &&
-            Array.isArray(this.submissionReceivedEmails)
-              ? this.submissionReceivedEmails
-              : [];
-          const response = await formService.createForm({
-            name: this.name,
-            description: this.description,
-            schema: this.formSchema,
-            identityProviders: generateIdps({
-              idps: this.idps,
-              userType: this.userType,
-            }),
-            showSubmissionConfirmation: this.showSubmissionConfirmation,
-            submissionReceivedEmails: emailList,
-          });
-          // Add the schema to the newly created default version
-          if (!response.data.versions || !response.data.versions[0]) {
-            throw new Error(
-              `createForm response does not include a form version: ${response.data.versions}`
-            );
+        if (this.formId) {
+          if (this.versionId) {
+            // If creating a new draft from an existing version
+            await this.schemaCreateDraftFromVersion();
+          } else if (this.draftId) {
+            // If updating an existing draft
+            await this.schemaUpdateExistingDraft();
```

Comment:
Not sure if making these new method wrappers is necessary. Those methods are not called anywhere else, so it would be sufficient for those contents to be in here directly. If however you are refactoring this to make unit testing potentially easier, this could be valid.

Reply:
Readability mostly, method was **way** too long with the 3 cases in it. Unit testing is the second thing to break it down for easier testing.